src/Reflection/Symbol.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace BetterReflection\Reflection;
+
+use PhpParser\Node;
+
+class Symbol
+{
+    const SYMBOL_CLASS = 'class';
+    const SYMBOL_FUNCTION = 'function';
+
+    /**
+     * @var string[]
+     */
+    private $validSymbols = [
+        self::SYMBOL_CLASS,
+        self::SYMBOL_FUNCTION,
+    ];
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    public function __construct($name, $type = self::SYMBOL_CLASS)
+    {
+        if ('\\' == $name[0]) {
+            $name = substr($name, 1);
+        }
+        $this->name = (string)$name;
```

Comment:
Consider validating `$name` against the regex `'/([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]+)(\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]+)*/'`, as that's what a valid class name can refer to.

Only one segment for functions/methods.

See https://github.com/Ocramius/ProxyManager/blob/7badedb5f8afaaccd496a096daadb19d7bf2b374/tests/ProxyManagerTest/Inflector/ClassNameInflectorTest.php#L127